cranelift/module/src/data_context.rs

```diff
@@ -46,6 +46,8 @@ pub struct DataDescription {
     pub function_relocs: Vec<(CodeOffset, ir::FuncRef)>,
     /// Data addresses to write at specified offsets.
     pub data_relocs: Vec<(CodeOffset, ir::GlobalValue, Addend)>,
+    /// Object file section
+    pub section: Option<(std::string::String, std::string::String)>,
```

Comment:
`std::string::String` is always in scope as `String` so style would be to just say `String` here. Also, we should note in the comment what each member of the pair is for - I think they are for segment and section, right? should we make the field named `custom_segment_section` to reflect that? Is there ever a use case where you'd want to specify a section but not a segment (maybe you're letting a linker script determine segments from section names?)

Reply:
For mach-O files the linker is always responsible for picking the right segment. In fact the object files should only contain a single segment.

Reply:
Thanks @bjorn3. So it sounds like we should be specifying a custom section and a custom segment in two different fields.

Reply:
@bjorn3 maybe i misunderstand but this:
seg: __OBJC
sec: __image_info

besides 
seg: __DATA
sec __data

isn't unusual on mach-o. (that said, segments are ignored on non mach-o afaik)

Reply:
> For mach-O files the linker is always responsible for picking the right segment. In fact the object files should only contain a single segment.

My understanding is that the linker uses the segment name specified in the object file. Note that segment names appear in two places: in the segment and in the section. The linker ignores the first of these, but still uses the segment name in the section. It has to do this, because `__const` can appear in both `__TEXT` and `__DATA`.